src/api/core/DatabaseServerlessHandler.ts

```diff
@@ -0,0 +1,31 @@
+import { ProxyResult } from "aws-lambda";
+import { Connection } from "typeorm";
+import { Database } from "../core/Database";
+
+
+export abstract class DatabaseServerlessHandler<T> {
+
+    protected connection: Connection;
+
+    protected abstract initializeDependencies(event?: T): void;
+
+    protected abstract onHandleEvent(event?: T): Promise<ProxyResult>;
+
+    public async execute(event: T): Promise<ProxyResult> {
+        this.initializeDatabase();
```

Comment:
@Aristevao this function call is wrong, because the `initializeDatabase()` function is a `async` function.